<a href="https://colab.research.google.com/github/ToheedAsghar/Smart_Meters_In_London_AI_Project/blob/main/Project_File.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Data Preparation


In [70]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.dates as dates
# % matplotlib inline
import seaborn as sns

In [71]:
# getting acorn details
acorn_details = pd.read_csv('/content/drive/MyDrive/SmartMeter_DataSet/acorn_details.csv', encoding='ISO-8859-1')

#displaying the structure of the acorn_details
acorn_details.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 826 entries, 0 to 825
Data columns (total 20 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   MAIN CATEGORIES  826 non-null    object 
 1   CATEGORIES       826 non-null    object 
 2   REFERENCE        825 non-null    object 
 3   ACORN-A          826 non-null    float64
 4   ACORN-B          826 non-null    float64
 5   ACORN-C          826 non-null    float64
 6   ACORN-D          826 non-null    float64
 7   ACORN-E          826 non-null    float64
 8   ACORN-F          826 non-null    float64
 9   ACORN-G          826 non-null    float64
 10  ACORN-H          826 non-null    float64
 11  ACORN-I          826 non-null    float64
 12  ACORN-J          826 non-null    float64
 13  ACORN-K          826 non-null    float64
 14  ACORN-L          826 non-null    float64
 15  ACORN-M          826 non-null    float64
 16  ACORN-N          826 non-null    float64
 17  ACORN-O         

In [72]:
#displaying first 5 rows
acorn_details.head()

MAIN CATEGORIES CATEGORIES  REFERENCE  ACORN-A  ACORN-B  ACORN-C  ACORN-D  \
0      POPULATION        Age    Age 0-4     77.0     83.0     72.0    100.0   
1      POPULATION        Age   Age 5-17    117.0    109.0     87.0     69.0   
2      POPULATION        Age  Age 18-24     64.0     73.0     67.0    107.0   
3      POPULATION        Age  Age 25-34     52.0     63.0     62.0    197.0   
4      POPULATION        Age  Age 35-49    102.0    105.0     91.0    124.0   

   ACORN-E  ACORN-F  ACORN-G  ACORN-H  ACORN-I  ACORN-J  ACORN-K  ACORN-L  \
0    120.0     77.0     97.0     97.0     63.0    119.0     67.0    114.0   
1     94.0     95.0    102.0    106.0     67.0     95.0     64.0    108.0   
2    100.0     71.0     83.0     89.0     62.0    104.0    459.0     97.0   
3    151.0     66.0     90.0     88.0     63.0    132.0    145.0    109.0   
4    118.0     93.0    102.0    103.0     76.0    111.0     67.0     99.0   

   ACORN-M  ACORN-N  ACORN-O  ACORN-P  ACORN-Q  
0    113.0     89.0    123.0    138.0    133.0  
1    116.0     86.0     89.0    136.0    106.0  
2     96.0     86.0    117.0    109.0    110.0  
3     96.0     90.0    140.0    120.0    120.0  
4     98.0     90.0    102.0    103.0    100.0

In [73]:
#displaying the last 5 rows
acorn_details.tail()

MAIN CATEGORIES                CATEGORIES                  REFERENCE  \
821    LEISURE TIME  Holiday Destination/Type                       Asia   
822    LEISURE TIME  Holiday Destination/Type  Activity / Outdoor Sports   
823    LEISURE TIME  Holiday Destination/Type                     Cruise   
824    LEISURE TIME  Holiday Destination/Type                    Package   
825    LEISURE TIME  Holiday Destination/Type              Self-catering   

     ACORN-A  ACORN-B  ACORN-C  ACORN-D  ACORN-E  ACORN-F  ACORN-G  ACORN-H  \
821    171.0    137.0     94.0    220.0    196.0     69.0    122.0     95.0   
822    298.0    278.0    138.0    119.0     93.0     96.0    113.0     96.0   
823    272.0    295.0    272.0     44.0     44.0     70.0     70.0     70.0   
824    196.0    186.0    166.0     49.0     49.0    101.0    101.0    121.0   
825    308.0    181.0    181.0     55.0     55.0    127.0    147.0    124.0   

     ACORN-I  ACORN-J  ACORN-K  ACORN-L  ACORN-M  ACORN-N  ACORN-O  ACORN-P  \
821     72.0     97.0    124.0     73.0     57.0     49.0     96.0    113.0   
822     84.0     96.0     56.0     48.0     48.0     46.0     23.0     23.0   
823     70.0     70.0     28.0     28.0     28.0     28.0     49.0     49.0   
824    106.0    101.0     48.0     75.0     75.0     75.0     54.0     54.0   
825    113.0    166.0     44.0     44.0     44.0     33.0     32.0     35.0   

     ACORN-Q  
821     63.0  
822     23.0  
823     49.0  
824     54.0  
825     35.0

In [74]:
#getting the information_households file
info_households = pd.read_csv('/content/drive/MyDrive/SmartMeter_DataSet/informations_households.csv', encoding='ISO-8859-1')

#displaying the structure of the infor_households
info_households.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5566 entries, 0 to 5565
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   LCLid          5566 non-null   object
 1   stdorToU       5566 non-null   object
 2   Acorn          5566 non-null   object
 3   Acorn_grouped  5566 non-null   object
 4   file           5566 non-null   object
dtypes: object(5)
memory usage: 217.5+ KB


In [75]:
#displaying the first 5 rows of the file
info_households.head()

LCLid stdorToU    Acorn Acorn_grouped     file
0  MAC005492      ToU   ACORN-        ACORN-  block_0
1  MAC001074      ToU   ACORN-        ACORN-  block_0
2  MAC000002      Std  ACORN-A      Affluent  block_0
3  MAC003613      Std  ACORN-A      Affluent  block_0
4  MAC003597      Std  ACORN-A      Affluent  block_0

In [76]:
#displaying the first 5 rows of the file
info_households.tail()

LCLid stdorToU    Acorn Acorn_grouped       file
5561  MAC002056      Std  ACORN-U       ACORN-U  block_111
5562  MAC004587      Std  ACORN-U       ACORN-U  block_111
5563  MAC004828      Std  ACORN-U       ACORN-U  block_111
5564  MAC001704      ToU  ACORN-U       ACORN-U  block_111
5565  MAC002774      ToU  ACORN-U       ACORN-U  block_111

In [77]:
# getting the weather_daily_darksky file
weather_daily = pd.read_csv('/content/drive/MyDrive/SmartMeter_DataSet/weather_daily_darksky.csv', encoding='ISO-8859-1')

#displaying the structure of the weather_daily_darksky
weather_daily.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 882 entries, 0 to 881
Data columns (total 32 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   temperatureMax               882 non-null    float64
 1   temperatureMaxTime           882 non-null    object 
 2   windBearing                  882 non-null    int64  
 3   icon                         882 non-null    object 
 4   dewPoint                     882 non-null    float64
 5   temperatureMinTime           882 non-null    object 
 6   cloudCover                   881 non-null    float64
 7   windSpeed                    882 non-null    float64
 8   pressure                     882 non-null    float64
 9   apparentTemperatureMinTime   882 non-null    object 
 10  apparentTemperatureHigh      882 non-null    float64
 11  precipType                   882 non-null    object 
 12  visibility                   882 non-null    float64
 13  humidity            

In [78]:
# displaying the front 5 rows of the file
weather_daily.head()

temperatureMax   temperatureMaxTime  windBearing                 icon  \
0           11.96  2011-11-11 23:00:00          123                  fog   
1            8.59  2011-12-11 14:00:00          198    partly-cloudy-day   
2           10.33  2011-12-27 02:00:00          225    partly-cloudy-day   
3            8.07  2011-12-02 23:00:00          232                 wind   
4            8.22  2011-12-24 23:00:00          252  partly-cloudy-night   

   dewPoint   temperatureMinTime  cloudCover  windSpeed  pressure  \
0      9.40  2011-11-11 07:00:00        0.79       3.88   1016.08   
1      4.49  2011-12-11 01:00:00        0.56       3.94   1007.71   
2      5.47  2011-12-27 23:00:00        0.85       3.54   1032.76   
3      3.69  2011-12-02 07:00:00        0.32       3.00   1012.12   
4      2.79  2011-12-24 07:00:00        0.37       4.46   1028.17   

  apparentTemperatureMinTime  ...  temperatureHigh          sunriseTime  \
0        2011-11-11 07:00:00  ...            10.87  2011-11-11 07:12:14   
1        2011-12-11 02:00:00  ...             8.59  2011-12-11 07:57:02   
2        2011-12-27 22:00:00  ...            10.33  2011-12-27 08:07:06   
3        2011-12-02 07:00:00  ...             7.36  2011-12-02 07:46:09   
4        2011-12-24 07:00:00  ...             7.93  2011-12-24 08:06:15   

   temperatureHighTime          uvIndexTime  \
0  2011-11-11 19:00:00  2011-11-11 11:00:00   
1  2011-12-11 14:00:00  2011-12-11 12:00:00   
2  2011-12-27 14:00:00  2011-12-27 00:00:00   
3  2011-12-02 12:00:00  2011-12-02 10:00:00   
4  2011-12-24 15:00:00  2011-12-24 13:00:00   

                                             summary   temperatureLowTime  \
0                             Foggy until afternoon.  2011-11-11 19:00:00   
1                  Partly cloudy throughout the day.  2011-12-12 07:00:00   
2                  Mostly cloudy throughout the day.  2011-12-27 23:00:00   
3  Partly cloudy throughout the day and breezy ov...  2011-12-02 19:00:00   
4                  Mostly cloudy throughout the day.  2011-12-24 19:00:00   

   apparentTemperatureMin  apparentTemperatureMaxTime  \
0                    6.48         2011-11-11 23:00:00   
1                    0.11         2011-12-11 20:00:00   
2                    5.59         2011-12-27 02:00:00   
3                    0.46         2011-12-02 12:00:00   
4                   -0.51         2011-12-24 23:00:00   

  apparentTemperatureLowTime moonPhase  
0        2011-11-11 19:00:00      0.52  
1        2011-12-12 08:00:00      0.53  
2        2011-12-28 00:00:00      0.10  
3        2011-12-02 19:00:00      0.25  
4        2011-12-24 20:00:00      0.99  

[5 rows x 32 columns]

For weather daily information we are going to use “temperatureMaxTime” (to extract the date), “temperatureMin” and “temperatureHigh” variables (to calculate the average temperature per day).
We also are going to add “Year”,“Year_month” and “weekend” variables to this data.

In [79]:
dates_data = pd.to_datetime(weather_daily['temperatureMaxTime'])
dates_data.dt.to_period('D')

0      2011-11-11
1      2011-12-11
2      2011-12-27
3      2011-12-02
4      2011-12-24
          ...    
877    2014-01-26
878    2014-02-27
879    2014-03-09
880    2014-02-12
881    2014-02-15
Name: temperatureMaxTime, Length: 882, dtype: period[D]

In [80]:
#labeling the days extracted now as weekend or weekdays
weekday = np.where(dates_data.dt.dayofweek < 5, 'Weekday', 'Weekend')

In [81]:
# converting the array to pd series object
weekday = pd.Series(weekday)
weekday

0      Weekday
1      Weekend
2      Weekday
3      Weekday
4      Weekend
        ...   
877    Weekend
878    Weekday
879    Weekend
880    Weekday
881    Weekend
Length: 882, dtype: object

In [82]:
# introducing a new variable "workday_weekend" in our weather_daily_darksky file
weather_daily['workday_Weekends'] = weekday
weather_daily.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 882 entries, 0 to 881
Data columns (total 33 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   temperatureMax               882 non-null    float64
 1   temperatureMaxTime           882 non-null    object 
 2   windBearing                  882 non-null    int64  
 3   icon                         882 non-null    object 
 4   dewPoint                     882 non-null    float64
 5   temperatureMinTime           882 non-null    object 
 6   cloudCover                   881 non-null    float64
 7   windSpeed                    882 non-null    float64
 8   pressure                     882 non-null    float64
 9   apparentTemperatureMinTime   882 non-null    object 
 10  apparentTemperatureHigh      882 non-null    float64
 11  precipType                   882 non-null    object 
 12  visibility                   882 non-null    float64
 13  humidity            

In [83]:
# introducing a new variable avg from temperature min and temperature max
weather_daily['avgTemperature'] = (weather_daily['temperatureMin'] + weather_daily['temperatureHigh'])/2
weather_daily.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 882 entries, 0 to 881
Data columns (total 34 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   temperatureMax               882 non-null    float64
 1   temperatureMaxTime           882 non-null    object 
 2   windBearing                  882 non-null    int64  
 3   icon                         882 non-null    object 
 4   dewPoint                     882 non-null    float64
 5   temperatureMinTime           882 non-null    object 
 6   cloudCover                   881 non-null    float64
 7   windSpeed                    882 non-null    float64
 8   pressure                     882 non-null    float64
 9   apparentTemperatureMinTime   882 non-null    object 
 10  apparentTemperatureHigh      882 non-null    float64
 11  precipType                   882 non-null    object 
 12  visibility                   882 non-null    float64
 13  humidity            

In [84]:
# displaying the average temperature
weather_daily['avgTemperature'].head()

0    9.860
1    5.535
2    9.180
3    4.960
4    5.550
Name: avgTemperature, dtype: float64

In [85]:
# introducing the new variable year for better understanding of the data year-wise
weather_daily['year'] = dates_data.map(lambda x: x.strftime('%Y'))

# similarly introducing a new variable year-month for more detail regarding months in years
weather_daily['year-month'] = dates_data.map(lambda x: x.strftime('%Y-%m'))

#displaying all new introduced variables
weather_daily[['workday_Weekends', 'avgTemperature', 'year', 'year-month']].head()

workday_Weekends  avgTemperature  year year-month
0          Weekday           9.860  2011    2011-11
1          Weekend           5.535  2011    2011-12
2          Weekday           9.180  2011    2011-12
3          Weekday           4.960  2011    2011-12
4          Weekend           5.550  2011    2011-12

For information_households table:


*   Acorn_grouped variables doesn't correctly match with acorn information that's why we need to map acorn information with correct acron groups.
*   Randomly Select 5 blocks (1, 10, 70, 75, 105) covering 5 main Acorn Group Information (Affluent, Rising, Comfortable, Stretched, Adversity).
*   These 5 datasets will be combined into one large dataset which will be named as "all_blocks".
*   Then user informations in information_households will be combined with "all_blocks" table.




In [86]:
# displaying information_households again
info_households.info()
info_households.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5566 entries, 0 to 5565
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   LCLid          5566 non-null   object
 1   stdorToU       5566 non-null   object
 2   Acorn          5566 non-null   object
 3   Acorn_grouped  5566 non-null   object
 4   file           5566 non-null   object
dtypes: object(5)
memory usage: 217.5+ KB


LCLid stdorToU    Acorn Acorn_grouped     file
0  MAC005492      ToU   ACORN-        ACORN-  block_0
1  MAC001074      ToU   ACORN-        ACORN-  block_0
2  MAC000002      Std  ACORN-A      Affluent  block_0
3  MAC003613      Std  ACORN-A      Affluent  block_0
4  MAC003597      Std  ACORN-A      Affluent  block_0

In [87]:
# getting unique accorn-letters
acorn_letter = info_households['Acorn'].unique()
acorn_letter

array(['ACORN-', 'ACORN-A', 'ACORN-B', 'ACORN-C', 'ACORN-D', 'ACORN-E',
       'ACORN-F', 'ACORN-G', 'ACORN-H', 'ACORN-I', 'ACORN-J', 'ACORN-K',
       'ACORN-L', 'ACORN-M', 'ACORN-N', 'ACORN-O', 'ACORN-P', 'ACORN-Q',
       'ACORN-U'], dtype=object)

In [88]:
# getting the 5 unique groups from acorn_groups
acorn_groups = info_households['Acorn_grouped'].unique()
acorn_groups

array(['ACORN-', 'Affluent', 'Comfortable', 'Adversity', 'ACORN-U'],
      dtype=object)

In [89]:
# mapping each acorn group with it's name
acorn_class = dict()
acorn = info_households['Acorn']

for al in acorn_letter:
  acorn_class[al] = info_households['Acorn_grouped'].iloc[acorn[acorn==al].index[0]]

acorn_class

{'ACORN-': 'ACORN-',
 'ACORN-A': 'Affluent',
 'ACORN-B': 'Affluent',
 'ACORN-C': 'Affluent',
 'ACORN-D': 'Affluent',
 'ACORN-E': 'Affluent',
 'ACORN-F': 'Comfortable',
 'ACORN-G': 'Comfortable',
 'ACORN-H': 'Comfortable',
 'ACORN-I': 'Comfortable',
 'ACORN-J': 'Comfortable',
 'ACORN-K': 'Adversity',
 'ACORN-L': 'Adversity',
 'ACORN-M': 'Adversity',
 'ACORN-N': 'Adversity',
 'ACORN-O': 'Adversity',
 'ACORN-P': 'Adversity',
 'ACORN-Q': 'Adversity',
 'ACORN-U': 'ACORN-U'}

In [90]:
# understanding the condition
acorn[acorn=='ACORN-B'].index[0]

159

In [91]:
# renaming the acorn_information with correct acorn_groups
acorn_class['ACORN-D'] = 'Rising'
acorn_class['ACORN-E'] = 'Rising'
acorn_class['ACORN-K'] = 'Stretched'
acorn_class['ACORN-L'] = 'Stretched'
acorn_class['ACORN-M'] = 'Stretched'
acorn_class['ACORN-N'] = 'Stretched'
acorn_class['ACORN-'] = 'NotPrivate_Houshold'
acorn_class['ACORN-U'] = 'NotPrivate_Houshold'

#implementing the change in info_households
for al, ac in acorn_class.items():
  info_households['Acorn_grouped'][info_households['Acorn']==al] = ac

In [92]:
# displaying the instance of the new dataframe
info_households[info_households['Acorn'] == 'ACORN-D']

LCLid stdorToU    Acorn Acorn_grouped      file
335  MAC004539      Std  ACORN-D        Rising   block_6
336  MAC000336      ToU  ACORN-D        Rising   block_6
337  MAC000342      ToU  ACORN-D        Rising   block_6
338  MAC001698      Std  ACORN-D        Rising   block_6
339  MAC000727      Std  ACORN-D        Rising   block_6
..         ...      ...      ...           ...       ...
622  MAC005205      Std  ACORN-D        Rising  block_12
623  MAC000291      Std  ACORN-D        Rising  block_12
624  MAC002670      Std  ACORN-D        Rising  block_12
625  MAC002144      Std  ACORN-D        Rising  block_12
626  MAC001787      Std  ACORN-D        Rising  block_12

[292 rows x 5 columns]

In [93]:
# displaying info_households again for revision of fields
info_households.info()
info_households.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5566 entries, 0 to 5565
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   LCLid          5566 non-null   object
 1   stdorToU       5566 non-null   object
 2   Acorn          5566 non-null   object
 3   Acorn_grouped  5566 non-null   object
 4   file           5566 non-null   object
dtypes: object(5)
memory usage: 217.5+ KB


LCLid stdorToU    Acorn        Acorn_grouped     file
0  MAC005492      ToU   ACORN-  NotPrivate_Houshold  block_0
1  MAC001074      ToU   ACORN-  NotPrivate_Houshold  block_0
2  MAC000002      Std  ACORN-A             Affluent  block_0
3  MAC003613      Std  ACORN-A             Affluent  block_0
4  MAC003597      Std  ACORN-A             Affluent  block_0

In [94]:
# moving towards the next step which was to randomly select 5 blocks

blocks = ["block_1", "block_10", "block_70", "block_75", "block_105"]

# selecting the record of only those blocks which we've selected
info_blocks = info_households[info_households['file'].isin(blocks)]

# displying the records
info_blocks

LCLid stdorToU    Acorn Acorn_grouped       file
50    MAC003840      Std  ACORN-A      Affluent    block_1
51    MAC003212      Std  ACORN-A      Affluent    block_1
52    MAC003182      Std  ACORN-A      Affluent    block_1
53    MAC003166      Std  ACORN-A      Affluent    block_1
54    MAC001628      Std  ACORN-A      Affluent    block_1
...         ...      ...      ...           ...        ...
5295  MAC001839      Std  ACORN-Q     Adversity  block_105
5296  MAC002045      Std  ACORN-Q     Adversity  block_105
5297  MAC002053      Std  ACORN-Q     Adversity  block_105
5298  MAC002054      Std  ACORN-Q     Adversity  block_105
5299  MAC002055      Std  ACORN-Q     Adversity  block_105

[250 rows x 5 columns]

In [95]:
# displaying the count of blocks falling in acorn_groups
info_blocks['Acorn_grouped'].value_counts()
# the sum of counts should be 250 as we get 250x5 previously

Acorn_grouped
Affluent       50
Rising         50
Comfortable    50
Stretched      50
Adversity      50
Name: count, dtype: int64

In [96]:
# moving towards 3rd step which was to get the blocks dataset and
# combine them into one big dataset

block_1 = pd.read_csv('/content/drive/MyDrive/SmartMeter_DataSet/daily_dataset/daily_dataset/block_1.csv', encoding='ISO-8859-1')
block_10 = pd.read_csv('/content/drive/MyDrive/SmartMeter_DataSet/daily_dataset/daily_dataset/block_10.csv', encoding='ISO-8859-1')
block_70 = pd.read_csv('/content/drive/MyDrive/SmartMeter_DataSet/daily_dataset/daily_dataset/block_70.csv', encoding='ISO-8859-1')
block_75 = pd.read_csv('/content/drive/MyDrive/SmartMeter_DataSet/daily_dataset/daily_dataset/block_75.csv', encoding='ISO-8859-1')
block_105 = pd.read_csv('/content/drive/MyDrive/SmartMeter_DataSet/daily_dataset/daily_dataset/block_105.csv', encoding='ISO-8859-1')

In [97]:
# creating a list of all dataframes
blocks = [block_1, block_10, block_70, block_75, block_105]

#conatenating them
all_blocks = pd.concat(blocks)

#displaying the structure of the all_blocks
all_blocks.info()

<class 'pandas.core.frame.DataFrame'>
Index: 157584 entries, 0 to 32097
Data columns (total 9 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   LCLid          157584 non-null  object 
 1   day            157584 non-null  object 
 2   energy_median  157580 non-null  float64
 3   energy_mean    157580 non-null  float64
 4   energy_max     157580 non-null  float64
 5   energy_count   157584 non-null  int64  
 6   energy_std     157064 non-null  float64
 7   energy_sum     157580 non-null  float64
 8   energy_min     157580 non-null  float64
dtypes: float64(6), int64(1), object(2)
memory usage: 12.0+ MB


In [98]:
# merging both the dataframes
all_blocks = pd.merge(all_blocks, info_households, on='LCLid')

#displaying the structure of the merged dataframe
all_blocks.info()

#displaying the top 5 records
all_blocks.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 157584 entries, 0 to 157583
Data columns (total 13 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   LCLid          157584 non-null  object 
 1   day            157584 non-null  object 
 2   energy_median  157580 non-null  float64
 3   energy_mean    157580 non-null  float64
 4   energy_max     157580 non-null  float64
 5   energy_count   157584 non-null  int64  
 6   energy_std     157064 non-null  float64
 7   energy_sum     157580 non-null  float64
 8   energy_min     157580 non-null  float64
 9   stdorToU       157584 non-null  object 
 10  Acorn          157584 non-null  object 
 11  Acorn_grouped  157584 non-null  object 
 12  file           157584 non-null  object 
dtypes: float64(6), int64(1), object(6)
memory usage: 15.6+ MB


LCLid         day  energy_median  energy_mean  energy_max  \
0  MAC000323  2012-03-06         0.5460     0.567700       0.908   
1  MAC000323  2012-03-07         0.2195     0.386062       1.475   
2  MAC000323  2012-03-08         0.1695     0.290937       1.086   
3  MAC000323  2012-03-09         0.2605     0.423812       1.587   
4  MAC000323  2012-03-10         0.2680     0.373813       1.222   

   energy_count  energy_std  energy_sum  energy_min stdorToU    Acorn  \
0            20    0.235788      11.354       0.138      Std  ACORN-A   
1            48    0.348265      18.531       0.054      Std  ACORN-A   
2            48    0.262756      13.965       0.056      Std  ACORN-A   
3            48    0.402155      20.343       0.068      Std  ACORN-A   
4            48    0.308453      17.943       0.061      Std  ACORN-A   

  Acorn_grouped     file  
0      Affluent  block_1  
1      Affluent  block_1  
2      Affluent  block_1  
3      Affluent  block_1  
4      Affluent  block_1

In [99]:
# introducing new variables (refer to ) for more understanding of data with dates
all_blocks['day'] = pd.to_datetime(all_blocks['day'])
all_blocks['year'] = pd.to_datetime(all_blocks['day']).dt.year
all_blocks['year-month'] = pd.to_datetime(all_blocks['day']).map(lambda x: x.strftime('%Y-%m'))

# deleting the rows with null value if any
all_blocks.dropna(inplace=True)

# displaying the structure of the
all_blocks.info()

<class 'pandas.core.frame.DataFrame'>
Index: 157064 entries, 0 to 157582
Data columns (total 15 columns):
 #   Column         Non-Null Count   Dtype         
---  ------         --------------   -----         
 0   LCLid          157064 non-null  object        
 1   day            157064 non-null  datetime64[ns]
 2   energy_median  157064 non-null  float64       
 3   energy_mean    157064 non-null  float64       
 4   energy_max     157064 non-null  float64       
 5   energy_count   157064 non-null  int64         
 6   energy_std     157064 non-null  float64       
 7   energy_sum     157064 non-null  float64       
 8   energy_min     157064 non-null  float64       
 9   stdorToU       157064 non-null  object        
 10  Acorn          157064 non-null  object        
 11  Acorn_grouped  157064 non-null  object        
 12  file           157064 non-null  object        
 13  year           157064 non-null  int32         
 14  year-month     157064 non-null  object        
dtypes: da

In [100]:
all_blocks['Acorn_grouped'].value_counts()

Acorn_grouped
Stretched      32770
Adversity      31951
Rising         31703
Affluent       31609
Comfortable    29031
Name: count, dtype: int64

<h2>3. Energy Consumption Analysis </h2>


<h3> 3.1 Monthly Energy Analysis throughout the Given Years </h3>